### PR TITLE
Review updates

### PR DESCRIPTION
**Pull request by dwsupplee**
Nice work on this Ajay :)

Do we have plans to implement a `Row` class? It seems like it could provide some value when working with results.

### CODE REVIEW

File: Bigtable/phpunit-conformance.xml.dist [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dwsupplee) [inline Bigtable/phpunit-conformance.xml.dist:1]: We use the term "conformance" for Firestore, so I moved the acceptance tests to match.

File: Bigtable/src/ChunkFormatter.php [hunk 2]
```diff
@@ -17,72 +17,119 @@
...
      *
-     * @return \Generator|mixed
-     * @throws \Google\Cloud\Bigtable\Exception\BigtableDataOperationException if it detects invalid state
+     * @return \Generator
+     * @throws BigtableDataOperationException If a malformed response is received.
      */
     public function readAll()
     {
-        foreach ($this->serverStream->readAll() as $readRowsResponse) {
+        // Chunks contain 3 properties:
```
Comment (dwsupplee) [inline Bigtable/src/ChunkFormatter.php:112]: I moved these notes down into the method body as the docblock gets exposed to users in our doc site, the notes seemed more aimed towards maintainers of the library.

File: Bigtable/src/ChunkFormatter.php [hunk 3]
```diff
@@ -198,57 +242,56 @@ private function commit()
...
-        $newRowkey = $chunk->getRowKey();
         $this->validateNewRow($chunk);
-        $this->rowKey = $newRowkey;
+        $this->rowKey = $chunk->getRowKey();
         $familyName = $chunk->getFamilyName()->getValue();
         $qualifierName = $chunk->getQualifier()->getValue();
-        $labels = ($chunk->getLabels()->getIterator()->valid())?
-            implode(iterator_to_array($chunk->getLabels()->getIterator())):'';
-        $timestamp = $chunk->getTimestampMicros();
+        $labels = ($chunk->getLabels()->getIterator()->valid())
```
Comment (dwsupplee) [inline Bigtable/src/ChunkFormatter.php:270]: Is there a reason we don't want to return the labels as an array?

File: Bigtable/src/DataClient.php [hunk 4]
```diff
@@ -103,6 +105,15 @@ public function __construct($instanceId, $tableId, array $config = [])
         if (isset($config['bigtableClient'])) {
             $this->bigtableClient = $config['bigtableClient'];
         } else {
+            // Temp workaround for large messages.
+            $config['transportConfig'] = [
```
Comment (dwsupplee) [inline Bigtable/src/DataClient.php]: I was testing some large messages and received some error messages about resource limits, this workaround addressed the issue. We can configure it more elegantly when we circle back on how the constructor options will work.

Comment (dwsupplee) [inline Bigtable/src/DataClient.php:109]: I was testing with large messages and received some error messages about resource limits, this workaround addressed the issue. We can configure it more elegantly when we circle back on how the constructor options will work.

File: Bigtable/src/DataClient.php [hunk 5]
```diff
@@ -199,67 +210,91 @@ public function upsert(array $rows, array $options = [])
...
+     * $row = $dataClient->readRow('jefferson');
+     *
+     * print_r($row);
+     * ```
+     *
+     * @param string $rowKey The row key to read.
+     * @param array $options [optional] Configuration options.
+     * @return array|null
+     */
+    public function readRow($rowKey, array $options = [])
```
Comment (dwsupplee) [inline Bigtable/src/DataClient.php:291]: Having a separate method for reading a single row seemed a little easier to reason about in my head than accepting either a string or array for `rowKeys` in `readRows`. WDYT?

File: Bigtable/src/Exception/BigtableDataOperationException.php [hunk 6]
```diff
@@ -17,14 +17,11 @@
 
 namespace Google\Cloud\Bigtable\Exception;
 
-use Google\Cloud\Core\Exception\GoogleException;
-
 /**
  * Exception thrown when a request fails.
  */
-class BigtableDataOperationException extends GoogleException
+class BigtableDataOperationException extends \Exception
```
Comment (dwsupplee) [inline Bigtable/src/Exception/BigtableDataOperationException.php:23]: We are planning to not extend the exceptions in `Core` anymore, so for now just having this extend the base exception.

File: Bigtable/src/ChunkFormatter.php (hunk 3 above)
Comment (ajaaym) [reply Bigtable/src/ChunkFormatter.php:270]: We need to talk with @sduskis I think currently server supports just one label.

File: Bigtable/src/DataClient.php (hunk 5 above)
Comment (ajaaym) [reply Bigtable/src/DataClient.php:291]: Thanks for putting this. We had a plan to add this after readRows.

**Comment by ajaaym** [comment]
@dwsupplee We were thinking about Row class, but other dynamic language like nodejs uses map like structure. Also php client for other product like mongodb uses map like structure.

**Pull request merged**